### PR TITLE
[Inductor] [Quant] Enable QLinear int8-mixed-bf16 Lowering

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #112486

**Summary**
- PR 7 for enabling Int8-Mixed-BF16 PT2E PTQ Quantization with Inductor https://github.com/pytorch/pytorch/issues/111640.
- Enable the QLinear int8-mixed-bf16 weight prepack and post grad lowering inside inductor.

**TestPlan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qlinear
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Comment by leslie-fang-intel** [comment]
Hi @jgong5, please help take look of this PR also.

**Review by jgong5 — APPROVED** [review]

File: torch/_inductor/fx_passes/quantization.py [hunk 1]
```diff
@@ -1147,33 +1151,41 @@ def _generate_qconv_weight_prepack_patterns(dtype=torch.float32):
...
+        to_fp32_node = sub_node.args[0]
+
+        assert to_fp32_node.target is prims.convert_element_type.default
+        assert sub_node.target is aten.sub.Tensor
+        assert mul_node.target is aten.mul.Tensor
+        if (
+            len(list(to_fp32_node.users)) != 1
+            or len(list(sub_node.users)) != 1
+            or len(list(mul_node.users)) != 1
+        ):
```
Comment (eellison) [inline torch/_inductor/fx_passes/quantization.py:1176]: I think you should be able to match this in the pattern.

**Review by eellison — APPROVED** [review]

File: test/inductor/test_mkldnn_pattern_matcher.py [hunk 2]
```diff
@@ -906,32 +906,40 @@ class M(torch.nn.Module):
...
 
             def forward(self, x):
-                return self.linear(x)
+                return self.linear2(self.linear(x))
 
         bias_list = [True, False]
-        for bias in bias_list:
-            mod = M(bias).eval()
-            v = torch.randn((2, 4))
+        for int8_mixed_bf16 in (
```
Comment (jerryzh168) [inline test/inductor/test_mkldnn_pattern_matcher.py:915]: separate tests might be cleaner

**Review by jerryzh168 — APPROVED** [review]

File: test/inductor/test_mkldnn_pattern_matcher.py (hunk 2 above)
Comment (leslie-fang-intel) [reply test/inductor/test_mkldnn_pattern_matcher.py:915]: Thanks for the suggestions and split the tests for fp32 and bf16.

**Pull request closed without merging**